dts/arm/st/stm32f1.dtsi

```diff
@@ -123,6 +123,15 @@
 			label = "UART_3";
 		};
 
+		usart4: serial@40004c00 {
```

Comment:
Looking in ext/hal/st/stm32cube/stm32f1xx/soc/stm32f1*.h files, it appears UART4 is present only in half of SoCs of STM32F1 series.
So, if you need UART4, you need to add it in the variant dsti file (stm32f103xe.dsti or stm32f107.dtsi)